src/Moq/Language/Flow/SetupSequencePhrase.cs

```diff
@@ -37,6 +37,14 @@ public ISetupSequentialAction Throws(Exception exception)
 			return this;
 		}
 
+		public ISetupSequentialAction Throws<TException>(Func<TException> exceptionFunction) where TException : Exception, new()
+		{
+			Guard.NotNull(exceptionFunction, nameof(exceptionFunction));
```

Comment:
I'm wondering, should we perhaps add `Guard.NotNull`s in the new `SetupPhrase` implementation methods, too?